### ISSUE TITLE
Improving handling of triple-quoted string .format(....)

### ISSUE THREAD

**Issue by peterjc** [ikamensh/flynt#38]
Flynt does not preserve multi-line triple-quoted strings and their given line breaks.

Instead it seemingly converts them into a long string literal with ``\n``, and uses that when calculating the maximum line length, and does not restore the new lines on output.

Example:

```python
length = 88
quotes = "double"

text = """\
# This is an example configuration
# Hash lines are comments

[flake8]
max-line-length={0}
inline-quotes="{1}"
""".format(length, quotes))

with open("config.ini", "w") as handle:
    handle.write(text)
```

With default settings, does nothing (presumably as it considered the line length with the triple quoted string converted into a single line):

```
$ flynt triples.py 
Running flynt v.0.40.1

Flynt run has finished. Stats:

Execution time:                            0.004s
Files modified:                            0

_-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_.
Please run your tests before commiting. Did flynt get a perfect conversion? give it a star at: 
~ https://github.com/ikamensh/flynt ~
Thank you for using flynt. Upgrade more projects and recommend it to your colleagues!
```

Forcing it with a longer max line length:

```
Running flynt v.0.40.1

Flynt run has finished. Stats:

Execution time:                            0.005s
Files modified:                            1
Character count reduction:                 7 (2.86%)

Per expression type:
No old style (`%`) expressions attempted.
`.format(...)` calls attempted:            1/1 (100.0%)
F-string expressions created:              1

_-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_.
Please run your tests before commiting. Did flynt get a perfect conversion? give it a star at: 
~ https://github.com/ikamensh/flynt ~
Thank you for using flynt. Upgrade more projects and recommend it to your colleagues!
```

Actual output:

```python
length = 88
quotes = "double"

text = f"""# This is an example configuration\n# Hash lines are comments\n\n[flake8]\nmax-line-length={length}\ninline-quotes="{quotes}\"\n"""

with open("config.ini", "w") as handle:
    handle.write(text)
```

Note that this still uses triple quotes, and it escaped one (but not both) of the quote characters in the template.

Desired output:

```python
length = 88
quotes = "double"

text = f"""\
# This is an example configuration
# Hash lines are comments

[flake8]
max-line-length={length}
inline-quotes="{quotes}"
"""

with open("config.ini", "w") as handle:
    handle.write(text)
```

**Comment by peterjc** [comment]
I'm guessing the fix could be here:

https://github.com/ikamensh/flynt/blob/3358ddf5ee5242a17210da64691d9b5a1d261091/src/flynt/transform/transform.py#L37

If dealing with triple quotes (``'''`` or ``"""`` style), treat ``\n`` specially. This would be a compromise as you would loose the difference between the following:

```python
text = f"""\
# This is an example configuration
# Hash lines are comments

[flake8]
max-line-length={length}
inline-quotes="{quotes}"
"""
```

and this somewhat eccentric version mixing literal new lines with ``\n`` as well:

```python
text = f"""\
# This is an example configuration
# Hash lines are comments\n
[flake8]
max-line-length={length}
inline-quotes="{quotes}"
"""
```

**Comment by ikamensh** [comment]
Acknowledged. From my early experiments with `astor` (used to restore AST to source code), it seems this is very doable. Will try to come up with an implementation.

**Comment by ikamensh** [comment]
Just sharing the joy of AST manipulation:
```
import ast
import astor

s = """
I'm Nobody! Who are you?{}
Are you – Nobody – too?
Then there's a pair of us!{}
Don't tell! they'd advertise – you know!

How dreary – to be – Somebody!{}
How public – like a Frog –
To tell one's name – the livelong June –
To an admiring Bog!
""".format(1, 2, 3)

with open(__file__) as f:
    txt = f.read()

from flynt.transform.FstringifyTransformer import FstringifyTransformer

tree = ast.parse(txt)
t = FstringifyTransformer()
t.visit(tree)

restored = astor.to_source(tree)
print(restored)
```

Output:
```
import ast
import astor
s = f"""
I'm Nobody! Who are you?{1}
Are you – Nobody – too?
Then there's a pair of us!{2}
Don't tell! they'd advertise – you know!

How dreary – to be – Somebody!{3}
How public – like a Frog –
To tell one's name – the livelong June –
To an admiring Bog!
"""
with open(__file__) as f:
    txt = f.read()
from flynt.transform.FstringifyTransformer import FstringifyTransformer
tree = ast.parse(txt)
t = FstringifyTransformer()
t.visit(tree)
restored = astor.to_source(tree)
print(restored)
```

**Comment by ikamensh** [comment]
This is fixed in 0.41 (imperfectly due to https://github.com/berkerpeksag/astor/issues/172) .

**Comment by peterjc** [comment]
Cheers! Confirming:

```python
length = 88
quotes = "double"

text = """\
# This is an example configuration
# Hash lines are comments

[flake8]
max-line-length={0}
inline-quotes="{1}"
""".format(length, quotes))

with open("config.ini", "w") as handle:
    handle.write(text)
```

```
$ flynt triples.py 
Running flynt v.0.42

Flynt run has finished. Stats:

Execution time:                            0.197s
Files modified:                            1
Character count reduction:                 13 (5.31%)

Per expression type:
No old style (`%`) expressions attempted.
`.format(...)` calls attempted:            1/1 (100.0%)
F-string expressions created:              1

_-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_.
Please run your tests before committing. Did flynt get a perfect conversion? give it a star at: 
~ https://github.com/ikamensh/flynt ~
Thank you for using flynt. Upgrade more projects and recommend it to your colleagues!

```

```python
length = 88
quotes = "double"

text = f"""# This is an example configuration
# Hash lines are comments

[flake8]
max-line-length={length}
inline-quotes="{quotes}\"
"""

with open("config.ini", "w") as handle:
    handle.write(text)
```

Happily the ``\"`` seems harmless in terms of code functionality (probably why no one identified that bug until now).

Thank you
